andela/hc-gnext
# #153114595 Set timeouts and grace values for checks for a maximum of 30 days

#### What does this PR do?
This PR extends the maximum timeout and grace period values for a created check from 30 days to a period of about 1 year.

#### Description of Task to be completed?
 - In the file `static/js/checks.js`, add two constants; `MONTH` and `YEAR` to represent month and year respectively.
- Still in the same file, change the `max` field of the `range` field _(present in both the periodSlider and graceSlider variables)_ to a value equal to the number of seconds in a year.
- Adjust the percentages used in the `range` object for a better visual representation of the sliders.
- This also makes it easy to set hourly, daily, and monthly checks.
- In the file `hc/front/forms.py` and in the `TimeoutForm` class, change the `max_value` field to the number of seconds in the year.
- Add the `YEAR` and `MONTH` variables to the Units to be utilised in `hc/front/templatetags/hc_extras.py` and set them to their appropriate values.

#### How should this be manually tested?
- This app is hosted on *Heroku* and can be accessed [here](https://hc-task-2.herokuapp.com)
- Enter a valid email address. A login link will be sent to the given email address.
- Click on the given link to create checks
- Checks with grace period and timeout period values greater than 30 days can be created by adjusting the sliders to the right.
-  Also, daily checks can be created by adjusting the sliders to the left.
- After creating the check, you can use **`curl`** to ping the provided **`Ping URL`** from the checks dashboard. This will alter the status of the check to either up, down or late.

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/story/show/153114595

#### Screenshots (if appropriate)
Originally, the maximum timeout and grace period for a created check was 30 days. 

<img width="1440" alt="screen shot 2017-12-09 at 10 29 05 am" src="https://user-images.githubusercontent.com/29917291/33817367-561672cc-de50-11e7-9245-e711117928f6.png">
<img width="1440" alt="screen shot 2017-12-09 at 10 29 14 am" src="https://user-images.githubusercontent.com/29917291/33817383-6337cf28-de50-11e7-8d7c-ceae603420ed.png">
<img width="1440" alt="screen shot 2017-12-09 at 10 29 18 am" src="https://user-images.githubusercontent.com/29917291/33817390-72a78cf0-de50-11e7-90bb-f70ef6243023.png">

After making our changes, the timeout and the grace period values were maximized to a period of 1 year.
- Before adjusting the grace period and timeout, the modal view was as shown below

<img width="1440" alt="screen shot 2017-12-09 at 10 26 21 am" src="https://user-images.githubusercontent.com/29917291/33817563-3ccf10f2-de51-11e7-961b-8a6ee656d18b.png">

- Adjusting the timeout and grace periods to the maximum
<img width="1440" alt="screen shot 2017-12-09 at 10 26 33 am" src="https://user-images.githubusercontent.com/29917291/33817608-68d11a9c-de51-11e7-99c1-7c4104149b44.png">

- The view of the check from the dashboard with its timeout and grace period values set to one year
<img width="1440" alt="screen shot 2017-12-09 at 10 26 38 am" src="https://user-images.githubusercontent.com/29917291/33817618-6ebc1b50-de51-11e7-9f36-9a863e8aa869.png">
